infor-design/enterprise
# 3389 - Fix iOS scrolling bug related to Modals

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR fixes an iOS bug where it would sometimes not be possible to scroll inside a scrollable page area after opening/closing multiple Modals, Messages, etc.

**Related github/jira issue (required)**:
Closes #3389

**Steps necessary to review your pull request (required)**:
Pull this branch, build, run the app.  Then try the following:

_To test the scrolling bugfix:_
- in Mobile Safari on an iOS device/simulator, open the Kitchen Sink page at http://localhost:4000
- Scroll down to the Modal examples and click "Show Modal".  The modal should appear.
- Close the modal, and scroll down to the Message samples beneath it.  Click any of the Message buttons.  The Message should appear.
- Close the message and attempt to scroll the Kitchen Sink page.  The page should scroll properly.
- Open Safari dev tools and inspect your device/simulator page. Ensure there is no `modal-engaged` CSS class on the `body` tag.

_Smoke test nested modals to make sure the overlay/container are still working properly when closing multiple items:_
- In any browser, open http://localhost:4000/components/modal/test-nested.html
- Open all three nested modals and close them.  Ensure the overlay only disappears after the highest-level modal is closed.

**Included in this Pull Request**:
